runtime/runtime_darwin.go

```diff
@@ -96,14 +98,22 @@ func WaitForLoginCompletion(timeout time.Duration) error {
 }
 
 func InteractiveUsername() (string, error) {
+	// The /usr/bin/last call is extremely expensive, and the logged in user
+	// does not change during lifetime of generic-worker process, so we can
```

Comment:
> Is this specifically because we reboot in between task runs?
> If we switched users without a reboot this would not be the case, right?

Yes, you are right - if the interactive console user changed without a reboot, the cached value would be invalid. This could be possible if for example instead of rebooting, we ran somthing like `killall loginwindow`, which would trigger the loginwindow to respawn, and log in as the next task user. We should certainly keep an eye on this.

> This seems like something that is specific to the way `generic-worker` is deployed on OSX as opposed to an absolute truth of how the code behaves.

At the moment the multiuser engine dictates that there is a reboot between tasks, which means that the generic-worker process has to stop and start. If there wasn't a reboot, we'd still be fine as long as we had a loop the spawned the worker process to run one task at a time, but if the worker process was long lived, and `loginwindow` was killed, it certainly wouldn't work in its current form.

> Given that this is a big win for our environment it's worth landing. I'm wondering if this could shoot us in the foot in the future if we change the way things are deployed/run in our infrastructure and forget about this.

We should definitely keep an eye on this, thank you for highlighting this. I think the main use case for not rebooting is so that we can run multiple workers on one machine in parallel, which would require that different task users use different desktops, so then I think we wouldn't be using the interactive desktop (a.k.a the console session), in which case we would have to rewrite a lot of the desktop integration code, including this method, so I'm hoping we would catch it then. However, if for some reason we wanted to stop rebooting, but still only running a single worker at a time using the console session, indeed this could be something that would need adapting.